### PR TITLE
Fixing message box for summaries subdialogue and adapting the dialogues. Fixes #2266

### LINKED ISSUE #2266: message box summaries sub dialog
In the column summaries, "choose summaries" sub dialog, a message box appears whenever I uncheck the last ticked checkbox to tell me I need to check at least 1 box. I can see the idea behind it, but it's quite annoying. I like to uncheck everything and then check the things I want, especially when using the dialog fo different things, so the messages comes up a lot.

I suggest we change it so that the message box only appears when you click return and you don't have anything checked. Then it just tells you OK won't be enabled until you check something, I think that's more reasonable, and less common. You should still be able to click Return but just not OK.

### PR DESCRIPTION
**Pull request by stevekogo**

### CODE REVIEW

File: instat/dlgColumnStats.vb [hunk 1]
```diff
@@ -134,6 +135,8 @@ Public Class dlgColumnStats
 
     Private Sub cmdSummaries_Click(sender As Object, e As EventArgs) Handles cmdSummaries.Click
         sdgSummaries.ShowDialog()
+        sdgSummaries.TestSummaries()
```
Comment (FrancoisJRenaud) [inline instat/dlgColumnStats.vb]: Could this message not be sent on the sdg as it is there that the user chooses the summary isn't it ? Maybe disable return if no summary selected ? Or is that something we usually don't do ?

File: instat/dlgDescribeOneVariable.vb [hunk 2]
```diff
@@ -31,11 +31,21 @@ Public Class dlgDescribeOneVariable
...
     Private Sub cmdSummaries_click(sender As Object, e As EventArgs) Handles cmdSummaries.Click
         sdgSummaries.ShowDialog()
+        sdgSummaries.TestSummaries()
+        TestOKEnabled()
     End Sub
 
     Public Sub TestOKEnabled()
         If Not ucrReceiverDescribeOneVar.IsEmpty() Then
-            ucrBaseDescribeOneVar.OKEnabled(True)
+            If chkCustomise.Checked Then
```
Comment (FrancoisJRenaud) [inline instat/dlgDescribeOneVariable.vb]: Can you try to make this If Else a bit more compact ?

File: instat/dlgDescribeTwoVariable.vb [hunk 3]
```diff
@@ -32,14 +32,17 @@ Public Class dlgDescribeTwoVariable
...
 
-    Private Sub cmdDisplayOptions_Click(sender As Object, e As EventArgs) Handles cmdDisplayOptions.Click
-        sdgDescribeDisplay.GrpBoxEnable()
-        sdgDescribeDisplay.ShowDialog()
-    End Sub
-
     Public Sub TestOKEnabled()
         If ((Not ucrReceiverFirstVar.IsEmpty()) And (Not ucrReceiverSecondVar.IsEmpty())) Then
-            ucrBaseDescribeTwoVar.OKEnabled(True)
+            If ((strVarType = "numeric" OrElse strVarType = "integer") AndAlso (strSecondVarType = "factor")) Then
```
Comment (FrancoisJRenaud) [inline instat/dlgDescribeTwoVariable.vb]: Could be written a bit more compact ? Also there is an And to convert to AndAlso just here...

Comment (FrancoisJRenaud) [inline instat/dlgDescribeTwoVariable.vb]: Also, should you not have the If chkSaveResult AndAlso "no name for result" then ok disabled ?

File: instat/sdgSummaries.vb [hunk 4]
```diff
@@ -87,4 +86,11 @@ Public Class sdgSummaries
     Private Sub grpsummaries_CheckedChanged(sender As Object, e As EventArgs) Handles chkNTotal.CheckedChanged, chkNonMissing.CheckedChanged, chkNMissing.CheckedChanged, chkMode.CheckedChanged, chkMean.CheckedChanged, chkRange.CheckedChanged, chkStdDev.CheckedChanged, chkMedian.CheckedChanged, chkSum.CheckedChanged, chkQuartiles.CheckedChanged, chkVariance.CheckedChanged, chkMaximum.CheckedChanged, chkMinimum.CheckedChanged
         SummariesParameters()
     End Sub
+
+    Public Sub TestSummaries()
+        SummariesParameters()
+        If strSummariesParameter = "c()" Then
+            MsgBox("OK will be disabled until you check atleast one summary.")
```
Comment (FrancoisJRenaud) [inline instat/sdgSummaries.vb]: in the text box: atleast should be at least. Also, maybe set it as MsgBox.Type OkOnly ?

File: instat/dlgColumnStats.vb (hunk 1 above)
Comment (stevekogo) [inline instat/dlgColumnStats.vb]: The sub-routine returns a textbox with a message when nothing is selected and then TeskOkEnabled will disable OK.

Comment (FrancoisJRenaud) [inline instat/dlgColumnStats.vb]: Yes nzuri, we don't disable return apparently

**Comment by FrancoisJRenaud** [comment]
I think it's ready for @dannyparsons  :)

**Pull request merged**